### PR TITLE
added kilowatthour to energy units

### LINKED ISSUE #1514: Add kilowatt-hour (kWh) to energy conversion units
**Problem Statement**

There is no easy way to convert energy values to kilowatt-hours.

**Evidence or User Insights**

The kilowatt-hour (kWh) is a common unit of energy, used for electricity bills and battery capacity. Outside of scientific and engineering circles, it is more popular than the joule (J). Another representation similar to kWh is the watt-second (W-s).

**Proposal**

An additional option to the energy conversion units for kilowatt-hour and watt-second would make conversions easier to other forms of energy.

**Goals**

- Add kilowatt-hour (kWh) and watt-second (W-s) to the list of energy conversions.
- Determine the common way to shorten kilowatt-hour and watt-second. kWh and W-s is simply how I have represented them here however they may not be the generally-accepted format.

**Non-Goals**

Add all possible combinations of metric prefixes and units of time (e.g., megawatt-minute, decajoules, etc.)

**Low-Fidelity Concept**

1 W-s is equivalent to 1 J. 1 kWh is equivalent to 3600 kJ or 3,600,000 J or 3,600,000 W-s.

**Requested Assignment**

If adding conversion units is relatively simple, I can implement this.

### PR DESCRIPTION
**Pull request by Flippchen**
### Description of the changes:
- Added Kilowatt-hour to energy units
- Added display name for english and german

I didn't added Ws because Joule is already implemented.

### How changes were validated:
- Added unit test

### CODE REVIEW

**Comment by Flippchen** [comment]
> 

@microsoft-github-policy-service agree

**Review by mfloto — APPROVED** [review]
LGTM

**Review by TardisKing40 — APPROVED** [review]
LGTM

**Review by MarionHinkel — APPROVED** [review]
LGTM

**Review by Nereuxofficial — DISMISSED** [review]
LGTM

**Review by asasine — APPROVED** [review]
Thanks for making this change!

**Review by Nereuxofficial — APPROVED** [review]
Looks good, didn't notice the typo tho

**Comment by Flippchen** [comment]
@tian-lt Hey, could I get some feedback on this PR?

File: src/Calculator/Resources/de-DE/Resources.resw
Comment (tian-lt) [inline src/Calculator/Resources/de-DE/Resources.resw:1]: No need to touch the resources other than `en-us/*`. We should focus on the `en-us/Resources.resw` and the localization works will be handled automatically.

File: src/Calculator/Resources/en-GB/Resources.resw
Comment (tian-lt) [inline src/Calculator/Resources/en-GB/Resources.resw:1]: Same here. Undo the changes of this file.

File: src/Calculator/Resources/en-US/Resources.resw [hunk 1]
```diff
@@ -1772,6 +1772,10 @@
   <data name="UnitAbbreviation_Joule" xml:space="preserve">
     <value>J</value>
     <comment>An abbreviation for a measurement unit of energy</comment>
+  </data>
+    <data name="UnitAbbreviation_Kilowatthour" xml:space="preserve">
+    <value>kWh</value>
+    <comment>An abbreviation for a measurement unit of energy</comment>
```
Comment (tian-lt) [inline src/Calculator/Resources/en-US/Resources.resw:1778]: nit: A more detailed description of the abbr. about ***Kilowatt-hour*** is much more appreciated.

**Review by tian-lt — CHANGES_REQUESTED** [review]

**Comment by tian-lt** [comment]
> @tian-lt Hey, could I get some feedback on this PR?

Thank you for sending out the PR.
The code looks good. Some fine-tuning is needed before we merge it into the main branch.
Also, we may need some other comments from the feature crew to see if the feature is good to go.

File: src/Calculator/Resources/en-US/Resources.resw [hunk 2]
```diff
@@ -2136,6 +2140,10 @@
   <data name="UnitName_Joule" xml:space="preserve">
     <value>Joules</value>
     <comment>A measurement unit for energy. (Please choose the most appropriate plural form to fit any number between 0 and 999,999,999,999,999)</comment>
+  </data>
+    <data name="UnitName_Kilowatthour" xml:space="preserve">
+    <value>Kilowatt-hours</value>
```
Comment (tian-lt) [inline src/Calculator/Resources/en-US/Resources.resw:2145]: I suppose it might be `Kilowatt hours`. The `Foot-pounds` may not be a good example here and we need to review the name again in future to see if it is a correct name.

Comment (dan-jac) [reply src/Calculator/Resources/en-US/Resources.resw:2145]: That Wikipedia article's Talk page is instructive.

From the debates there, it seems as if common usage is to write "Kilowatt-hours" and "kWh".
Strict technical usage, as seen in IEC and NIST documents, is to write "Kilowatt hours" and "kW·h".

We certainly haven't adopted the strictly-correct abbreviation "kW·h" (for good reason – no one really uses this), so I'd say we should also stick with common usage of the long name: "Kilowatt-hours".

**Comment by Flippchen** [comment]
@tian-lt Thanks for your review. I have made changes according to your comments. I kept the name `Kilowatt-hour`

**Comment by tian-lt** [comment]
@Flippchen 
Looks good to me. Thank you for making this feature. I'm going to approve this PR and we will see this new unit available in Calculator in the upcoming releases.

**Review by tian-lt — DISMISSED** [review]

**Comment by tian-lt** [comment]
Hi @TardisKing40, looks like this PR is pending for your review. Could you take a look at the changes and leave any comments if you have?

**Comment by TardisKing40** [comment]
Looks good 👍

**Review by TardisKing40 — APPROVED** [review]

**Comment by tian-lt** [comment]
Hmm... the pipeline has been paused for this PR and seems it cannot resume automatically. @Flippchen do you mind republish this PR (close and open again)? Maybe it can help trigger the pipeline correctly. 😓
image

**Comment by Flippchen** [comment]
@tian-lt I closed and reopened the PR, but it did not help. I changed the branch. It needs your review now, but hopefully it helped.

**Review by tian-lt — APPROVED** [review]

**Comment by Flippchen** [comment]
@tian-lt Should I open a whole new PR?

**Comment by tian-lt** [comment]
> @tian-lt Should I open a whole new PR?

It is weird to to see such case. Let’s try to open another PR but keep this open as well. I’ll try to fix it one the ADO side.

**Review by tian-lt — APPROVED** [review]

**Comment by tian-lt** [comment]
Hi @mcooley, could you help take a look at this? It's hard to trigger *Calculator-CI* for this PR.

**Comment by hanzhang54** [comment]
> Hi @mcooley, could you help take a look at this? It's hard to trigger _Calculator-CI_ for this PR.

We made some changes to the CI pipeline before for security concern. Now it requires a comment like "/azp run" from team member before building a PR.
